src/renderer/components/+catalog/catalog-entity-details.tsx

```diff
@@ -68,8 +68,12 @@ export class CatalogEntityDetails<T extends CatalogEntity> extends Component<Pro
                 material={item.entity.spec.icon?.material}
                 background={item.entity.spec.icon?.background}
                 disabled={!item?.enabled}
-                onClick={() => item.onRun(catalogEntityRunContext)}
-                size={128} />
+                onClick={() => {
+                  this.props.onClickDetailPanelIcon(item);
```

Comment:
I don't think we should pass this in as props, we should have an unexported function that takes an `CatalogEntity` and does the work once (DRY as it were).

Reply:
Can the refactoring be another PR?